python/tvm/autotvm/task/task.py

```diff
@@ -0,0 +1,340 @@
+            The tvm schedule
+        arg_bufs: Array of tvm.tensor.Tensor
+            The input/output buffers
+        """
+        config.flop = 0
+        with ApplyConfig(config):
+            sch, arg_bufs = self.func(*self.args, **self.kwargs)
+        if not self.flop:
+            self.flop = config.flop or compute_flop(sch)
+        return sch, arg_bufs
+
+    def __repr__(self):
+        return "Task(func_name=%s, args=%s, kwargs=%s, workload=%s)" % (
+            self.name, self.args, self.kwargs, self.workload
+        )
+
+
+TASK_TABLE = {
+}
+
+def register(name, func=None, override=False):
+    """Register a task function.
+
+    Parameters
+    ----------
+    name : str
+        The name to identify the task.
+    func : callable
+        The function to be registered.
+    override : bool
+        Whether override existing registration.
+
+    Returns
+    -------
+    func: callable
+        The registered function
+    """
+    def _do_reg(myf):
+        if name in TASK_TABLE and not override:
+            raise ValueError(
+                "Key %s is already registered" % name)
+        TASK_TABLE[name] = myf
+        return myf
+    if func:
+        return _do_reg(func)
+    return _do_reg
+
+
+def create(func_name, args):
+    """Create a new tuning task.
+
+    Parameters
+    ----------
+    func_name : str or callable
+        The task function
+    args : tuple
+        Positional arguments for the task function
+    """
+    if callable(func_name):
+        func = func_name
+        func_name = func.func_name if hasattr(func, 'func_name') else func.__name__
+        if func_name in TASK_TABLE:
+            assert func == TASK_TABLE[func_name], "Find name conflict in task registration. " \
+                                                  "Consider to choose another name for this task"
+        else:
+            register(func_name, func=func)
+    func = TASK_TABLE[func_name]
+    return Task(func_name, func, args)
+
+def create_task(func_name, args, target, target_host=None):
+    """Create a new task and initialize its config space.
+    This is a shortcut for :any:`task.create` + :any:`Task.init_space`.
+    Note that kwargs is not supported
+
+    Parameters
+    ----------
+    func_name : str or callable
+        The task function
+    args : tuple
+        Positional arguments
+    target : Target
+        The compilation target
+    target_host: Target, optional
+        The compilation target for host side
+
+    Returns
+    -------
+    tsk: Task
+        a task object
+    """
+    ret = create(func_name, args)
+    ret.init_space(target, target_host=target_host)
+
+    return ret
+
+def simple_template(func):
+    """
+    decorate a function as a template
+
+    Parameters
+    ----------
+    func: callable
+        A callable template function.
+        Its argument should be hashable values.
+        Its return value should be a Tuple(Schedule, Array of Tensor)
+
+    Returns
```

Comment:
need an example block here.

simple_template deserves its own file. 
Need to explain why it is not in template